src/cmap/manager.rs

```diff
@@ -38,17 +38,11 @@ impl PoolManager {
     }
 
     /// Mark the pool as "ready" as per the CMAP specification.
-    pub(super) async fn mark_as_ready(&self) {
-        let (message, acknowledgment_receiver) = AcknowledgedMessage::package(());
-        if self
-            .sender
-            .send(PoolManagementRequest::MarkAsReady {
-                _completion_handler: message,
-            })
-            .is_ok()
-        {
-            acknowledgment_receiver.wait_for_acknowledgment().await;
-        }
+    ///
+    /// Since management requests are treated with the highest priority by the pool and will be
```

Comment:
I updated the pool worker task to use a biased `select!`, which means that it will poll in a certain order. Specifically for this, I updated it to check for management requests (clearing, marking as ready, etc) before checking for check-out requests. This means that we don't actually have to wait for acknowledgment that the pool is ready before returning here, since we know that no check-out requests will be processed before this `MarkAsReady` is.